stellarphot/core.py

```diff
@@ -30,6 +30,36 @@
 # https://docs.gammapy.org/dev/_modules/gammapy/analysis/config.html
 
 
+class UnitType(Unit):
+    # Validator for Unit type
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v):
+        return Unit(v)
+
+    @classmethod
+    def __modify_schema__(cls, field_schema, field):
+        # Set default values for the schema in case the field doesn't provide them
```

Comment:
I am honestly not sure how to assess what is going on here with `__modify_schema__`, need to dig into pydantic again or trust your tests. :). I will note `__modify_schema__` doesn't appear to be supported in v2 of pydantic (based on https://docs.pydantic.dev/latest/errors/usage_errors/#custom-json-schema).

Reply:
To the extent I understand it, this is providing some of the information needed to generate a json schema for types that are either not built in to Python or pydantic. 

I wish ipyautoui had already made the 1 to 2 transition so we could just code for 2. It looks like there is a replacement for `__modify_schema__` in 2.